### ISSUE TITLE
FAILED tests/test_deep_learning4e.py::test_rnn - AttributeError: module 'keras.preprocessing.sequence'

### ISSUE THREAD

**Issue by ailabuser** [aimacode/aima-python#1262]
FAILED tests/test_deep_learning4e.py::test_rnn - AttributeError: module 'keras.preprocessing.sequence' has no attribute 'pad_sequences'

Need to replace 'keras.preprocessing.sequence' with 'keras_preprocessing.sequence'

Python 3.8.13, Tensorflow 2.10.0, Keras 2.10.0

**Comment by EnthusiasticTeslim** [comment]
Yes, the py.test failed because of this issue. It is noteworthy to mention that the attribute has been moved to keras.utils in keras 2.9....In your deep_learning4e.py file, replace 

`from keras.preprocessing import sequence`

with 

`from keras.utils import pad_sequences`

Cheers,
Teslim.
<img width="847" alt="image" src="https://user-images.githubusercontent.com/36744823/196042022-3d1ec617-07eb-4645-afff-ec9175f59bb5.png">

**Comment by liukkcggvb** [comment]
> Yes, the py.test failed because of this issue. It is noteworthy to mention that the attribute has been moved to keras.utils in keras 2.9....In your deep_learning4e.py file, replace
> 
> `from keras.preprocessing import sequence`
> 
> with
> 
> `from keras.utils import pad_sequences`
> 
> Cheers, Teslim. <img alt="image" width="847" src="https://user-images.githubusercontent.com/36744823/196042022-3d1ec617-07eb-4645-afff-ec9175f59bb5.png">

Your answer helped me, to add, let's remove "sequences." from all sequence.pad_sequences in the file.

**Comment by talessl** [comment]
thank u so much for help me, both of u!

**Comment by dmeoli** [comment]
tests/test_deep_learning4e.py (including test_rnn) passes on current master (verified locally). The Keras code was modernized. Closing as resolved.
